### DISCUSSION TITLE
How to get statistics of column usage of a table

### DISCUSSION THREAD

**Discussion by sky9611** [ClickHouse/ClickHouse#51463] [Q&A] [answered]
Is it possible to determine in ClickHouse how many times a column has been selected or used as a filter? This kind of statistic could assist in optimizing a wide table, such as removing unused columns or adding frequently used columns as new orderBy keys.

Currently, the only solution I have found is to utilize the `system.query_log` table for analyzing the SQL queries, but it is not very efficient. I am curious if there are any other best practices available.

**Comment by vadimp** [comment, marked as answer]
From the docs [query_log](https://clickhouse.com/docs/en/operations/system-tables/query_log) docs:
```
columns (Array(LowCardinality(String))) — Names of the columns present in the query.
```

Contains all the columns used in the query already parsed.

>  How many times a column has been selected or used as a filter

"Used columns" vs "filtered by columns" are not currently tracked.
